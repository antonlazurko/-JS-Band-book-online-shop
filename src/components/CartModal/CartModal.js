/* eslint-disable import/no-unresolved */
import { useDispatch } from 'react-redux';
import { cartActions } from 'redux/cart';

const CartModal = () => {
  const dispatch = useDispatch();
  return (
    <button
      type="button"
      onClick={() => {
        console.log('click');
        dispatch(cartActions.cartModalClose());
      }}
    ></button>
  );
};

export default CartModal;
