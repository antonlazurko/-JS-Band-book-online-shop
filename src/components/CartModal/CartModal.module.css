.authModalBackdrop {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: 999;
  display: flex;
  justify-content: center;
  align-items: center;
background-color: rgba(0, 0, 0, 0.56);
}
.authModal {
  position: relative;
  width: 500px;
}