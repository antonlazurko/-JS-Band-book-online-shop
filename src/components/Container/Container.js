import React from 'react';

const Container = ({ children }) => (
  <div className="container" style={{ marginTop: '30px' }}>
    {children}
  </div>
);
export default Container;
