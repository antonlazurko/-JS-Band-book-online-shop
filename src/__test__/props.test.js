import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Switch, Route, Redirect, BrowserRouter } from 'react-router-dom';
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { render } from '@testing-library/react';
import CartModal from '../components/CartModal/CartModal';
import Container from '../components/Container/Container';

configure({ adapter: new Adapter() });

describe('CartModal', () => {
  const initialState = {
    authorization: { userName: 'anton' },
    cart: { cart: [] },
  };
  const cartInfo = [
    {
      bookId: '1',
      count: 2,
      title: 'string',
      price: 15,
    },
  ];
  const mockStore = configureStore();

  test('should render CartModal', () => {
    const store = mockStore(initialState);
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Route>
            <CartModal cartInfo={cartInfo} />
          </Route>
        </BrowserRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
describe('Container', () => {
  const initialState = {
    authorization: { userName: 'anton' },
    cart: { cart: [] },
  };
  const mockStore = configureStore();

  test('should render Container', () => {
    const store = mockStore(initialState);
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Route>
            <Container children={<div>I am children</div>} />
          </Route>
        </BrowserRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
