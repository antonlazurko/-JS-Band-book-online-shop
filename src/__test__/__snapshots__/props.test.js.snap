// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartModal should render CartModal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div
            class="modal-dialog"
          >
            <div
              class="modal-content"
            >
              <div
                class="modal-header"
              >
                <p
                  class="modal-title"
                >
                  You successfully placed on order!
                </p>
              </div>
              <div
                class="modal-body"
              >
                <div>
                  <table
                    border="1px"
                    class="table"
                  >
                    <thead>
                      <tr>
                        <th>
                          Name
                        </th>
                        <th>
                          Count
                        </th>
                        <th>
                          Price
                        </th>
                        <th>
                          Total
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td>
                          string
                        </td>
                        <td>
                          2
                        </td>
                        <td>
                          15
                        </td>
                        <td>
                          30
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  <div>
                    Total price:
                    30
                    $
                  </div>
                </div>
              </div>
              <div
                class="modal-footer"
              >
                <button
                  class="btn btn-outline-secondary"
                  type="button"
                >
                  Close
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div
          class="modal-dialog"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <p
                class="modal-title"
              >
                You successfully placed on order!
              </p>
            </div>
            <div
              class="modal-body"
            >
              <div>
                <table
                  border="1px"
                  class="table"
                >
                  <thead>
                    <tr>
                      <th>
                        Name
                      </th>
                      <th>
                        Count
                      </th>
                      <th>
                        Price
                      </th>
                      <th>
                        Total
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        string
                      </td>
                      <td>
                        2
                      </td>
                      <td>
                        15
                      </td>
                      <td>
                        30
                      </td>
                    </tr>
                  </tbody>
                </table>
                <div>
                  Total price:
                  30
                  $
                </div>
              </div>
            </div>
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-outline-secondary"
                type="button"
              >
                Close
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Container should render Container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        style="margin-top: 30px;"
      >
        <div>
          I am children
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      style="margin-top: 30px;"
    >
      <div>
        I am children
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
